src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareLanguageServerClient.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+using Nerdbank.Streams;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    [ContentType("C#_LSP")]
+    [ContentType("VB_LSP")]
+    [Export(typeof(ILanguageClient))]
+    internal class LiveShareLanguageServerClient : ILanguageClient
+    {
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly IThreadingContext _threadingContext;
+        private readonly LanguageServerProtocol _languageServerProtocol;
+        private readonly Workspace _workspace;
+
+        /// <summary>
+        /// Gets the name of the language client (displayed to the user).
+        /// </summary>
+        public string Name => ServicesVSResources.LiveShare_CSharp_Visual_Basic_Language_Server_Client;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public IEnumerable<string>? ConfigurationSections { get; } = null;
+
+        /// <summary>
+        /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
+        /// This may be null if the client does not need custom initialization options.
+        /// </summary>
+        public object? InitializationOptions { get; } = null;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public IEnumerable<string>? FilesToWatch { get; } = null;
+
+        public event AsyncEventHandler<EventArgs>? StartAsync;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public event AsyncEventHandler<EventArgs>? StopAsync { add { } remove { } }
```

Comment:
why is it unused

Reply:
We'll stop the server when VS shuts down.  We could potentially do something fancy to start/stop when a user starts a liveshare session, but I don't think we'd get much of a gain from doing something like that.